lib/reservation_maker.rb

```diff
@@ -0,0 +1,93 @@
+require_relative 'reservation'
+require_relative 'date_range'
+require_relative 'block'
+class ReservationError < StandardError
+end
+
+module Hotel
+  class Reservation_Maker
+    attr_reader :rooms 
+    attr_accessor :reservations, :blocks
+    
+    def initialize
+      rooms = Array(1..20)
+      @rooms = rooms
+      @reservations = []
+      @blocks = []
+    end 
+
+    def reserve_from_block(check_in, check_out)
+      date_range = Hotel::Date_Range.new(check_in, check_out)
+      @blocks.each do |block|
+        if !block.date_range.include?(check_in) 
+          raise ReservationError, "Date range is not included in existing block!"
```

Comment:
Hm, this code reads to me like "Check every reservation. If the date_range made above with our proposed check_in and check_out overlaps with any reservation's check_in and check_out, then we should raise a ReservationError." Does this mean that an error will be raised if Room No. 1 has an overlapping reservation, even if Rooms 2-20 are available? In either case, this section of the code doesn't have a unit test, so it's hard for me to check at the moment.